scripts/apps/search/directives/ItemList.js

```diff
@@ -668,6 +668,147 @@ export function ItemList(
                 }
             });
 
+            var MarkedDesksList = React.createClass({
```

Comment:
Same here, can we use ES6 classes, JSX, and document all of this? Also, please put this in a separate file. `ItemList` is already one of our biggest files and it's begging for a refactor :)